lua/hbac/telescope/init.lua

```diff
@@ -0,0 +1,93 @@
+local M = {}
+
+local hbac_config = require("hbac.setup").opts
+local make_display = require("hbac.telescope.make_display")
+local state = require("hbac.state")
+local subcommands = require("hbac.command.subcommands")
+local utils = require("hbac.utils")
+
+local action_state = require("telescope.actions.state")
+local finders = require("telescope.finders")
+local pickers = require("telescope.pickers")
+local telescope_conf = require("telescope.config").values
+
+local display = make_display.display
+
+local get_entries = function()
+	local entries = {}
+	local buflist = utils.get_buffers()
+	for _, bufnr in ipairs(buflist) do
+		local bufname = vim.api.nvim_buf_get_name(bufnr)
+		table.insert(entries, {
+			display = display,
+			value = bufnr,
+			ordinal = bufname,
+		})
+	end
+	return entries
+end
+
+local make_finder = function()
+	return finders.new_table({
+		results = get_entries(),
+		entry_maker = function(entry)
+			return {
+				value = entry.value,
+				display = entry.display,
+				ordinal = entry.ordinal,
+				path = entry.ordinal,
+			}
+		end,
+	})
+end
+
+local execute_telescope_command = function(prompt_bufnr, cmd_func)
+	local picker = action_state.get_current_picker(prompt_bufnr)
+	local multi_selection = picker:get_multi_selection()
+
+	if next(multi_selection) then
+		for _, entry in ipairs(multi_selection) do
+			cmd_func(entry.value)
+		end
+	else
+		local single_selection = action_state.get_selected_entry()
+		cmd_func(single_selection.value)
+	end
+	picker:refresh(make_finder(), { reset_prompt = true })
+end
+
+local attach_mappings = function(prompt_bufnr, map)
+	local telescope_commands = {
+		close_unpinned = subcommands.close_unpinned,
+		delete_buffer = hbac_config.close_command,
+		toggle_all = subcommands.pin_all,
+		toggle_selections = state.toggle_pin,
+	}
+
+	for command, mappings in pairs(hbac_config.telescope.mappings) do
+		local modes = { "n", "i" }
+		for i, mapping in ipairs(mappings) do
+			local mode = modes[i]
+			map({ mode }, mapping, function()
+				execute_telescope_command(prompt_bufnr, telescope_commands[command])
+			end, { noremap = true, silent = true })
+		end
+	end
+
+	return true
+end
+
+M.pin_picker = function(opts)
+	opts = opts or {}
+	pickers
+		.new(opts, {
+			prompt_title = "Hbac Pin States",
+			finder = make_finder(),
+			sorter = telescope_conf.generic_sorter(opts),
+			attach_mappings = attach_mappings,
+			previewer = telescope_conf.file_previewer(opts),
+		})
+		:find()
+end
```

Comment:
I wasn't sure how to interpret the request to make the picker more like the buffer picker beyond letting the default action (enter) open the selected buffer in a window. This is what my buffer picker looks like:

<img width="800" alt="SCR-20230515-tfdt" src="https://github.com/axkirillov/hbac.nvim/assets/23170004/3559d6df-a835-4cba-85fc-4ba529e163d6">

This is my first time making a Telescope picker so any advice and criticism is appreciated. What other changes would you like to see?